src/status_im2/contexts/wallet/events.cljs

```diff
@@ -10,3 +56,74 @@
   {:events [:wallet/clean-scanned-address]}
   [{:keys [db]}]
   {:db (dissoc db :wallet/scanned-address)})
+
+(defn fetch-address-suggestions
+  [{:keys [db]} [address]]
+  {:db (assoc db
+              :wallet/local-suggestions
+              (cond
+                (= address
+                   (get-in
+                    address-local-suggestion-saved-contact-address-mock
+                    [:accounts 0 :address]))
+                [address-local-suggestion-saved-contact-address-mock]
+                (= address
+                   (get address-local-suggestion-saved-address-mock
+                        :address))
+                [address-local-suggestion-saved-address-mock]
+                :else (find-matching-addresses address))
+              :wallet/valid-ens-or-address?
+              false)})
+
+(re-frame/reg-event-fx :wallet/fetch-address-suggestions fetch-address-suggestions)
```

Comment:
I think we agreed to put effects in a separate file `effects.cljs`

Reply:
Actually these are events? maybe ignore that comment 😅 

however, I do think we should not really be using `re-frame.core` in these files and the tooling comes from `utils.re-frame` in 99% of the cases. 

also, I thought we wanted to do 
```
(rf/reg-event-fx :wallet/fetch-address-suggestions
 (fn [{:keys [db]} ...]
  ...
```
perhaps @ilmotta can advise as it seems we should clarify the approach since not using `rf/defn` anymore

or why have you gone with this approach @briansztamfater 🤔

Reply:
Actually is not `rf/defn` but `defn`. I think it is the same, but defining a named function outside of the event registration.

For context, I tried to copy some recent example, for example this one in activity center domain https://github.com/status-im/status-mobile/blob/6f9bcd1bb1754f7735d33e966717c82e32415c2f/src/status_im2/contexts/shell/activity_center/events.cljs#L40

Happy to modify the approach if necessary!

Reply:
@J-Son89, @briansztamfater, the reason why we sometimes need to defn event handlers is to unit test them or if we need to call them in places that still use `rf/merge` (there are quite a few). The example @briansztamfater gave is written that way because the function `open-activity-center` has a unit test. I haven't started the refactor to the new/simplified way, but it's really great to see it's getting traction. Thanks guys